factom_core/block_elements/chain_commit.py

```diff
@@ -13,25 +13,37 @@ class ChainCommit:
     entry_hash: bytes
     ec_spent: int
     ec_public_key: bytes
-    signature: bytes
+    signature: bytes = None
 
     def __post_init__(self):
         # TODO: value assertions
         pass
 
+    def set_signature(self, signature: bytes):
+        assert isinstance(signature, bytes)
+        self.signature = signature
+
     def marshal(self):
         """Marshals the ChainCommit according to the byte-level representation shown at
         https://github.com/FactomProject/FactomDocs/blob/master/factomDataStructureDetails.md#chain-commit
         """
+        # Fail if signature is not set.
+        assert self.signature
```

Comment:
I'd prefer if this followed the python convention for none-value checking with `self.signature is not None`, as you already have it in the entry commit file

Reply:
That's what I had earlier I just misunderstood your previous comment, apologies.